deployer/troubleshooting-integration/src/main/java/io/hyscale/troubleshooting/integration/actions/FixCrashingApplication.java

```diff
@@ -64,15 +65,17 @@ public void process(TroubleshootingContext context) {
 		} else {
 			V1ContainerState v1ContainerState = PodStatusUtil.getLastState(pod);
 			Integer statusCode = PodStatusUtil.getExitCode(v1ContainerState);
-			PodStatusCode.Signals signals = PodStatusCode.Signals.fromCode(statusCode);
-			if (signals != null) {
-				report.setReason(AbstractedErrorMessage.SERVICE_COMMANDS_FAILURE.formatReason(signals.getSignal()));
-			} else {
-				report.setReason(AbstractedErrorMessage.APPLICATION_CRASH.getReason());
+			if (statusCode!=null) {
+				PodStatusCode.Signals signals = PodStatusCode.Signals.fromCode(statusCode);
+				String exitCode = (signals != null) ? signals.getSignal() : EXIT_CODE + statusCode.toString();
+				report.setReason(AbstractedErrorMessage.SERVICE_COMMANDS_FAILURE.formatReason(exitCode));
+			}else {
+				report.setReason(AbstractedErrorMessage.SERVICE_COMMANDS_FAILURE.getMessage());
```

Comment:
Should set APPLICATION_CRASH.getReason() nah ?
Please verfiy this once, i think the previous one is correct